dask/tests/test_distributed.py

```diff
@@ -794,7 +794,7 @@ def test_set_index_no_resursion_error(c):
         pytest.fail("dd.set_index triggered a recursion error")
 
 
-@pytest.mark.xfail(reason="https://github.com/dask/dask/issues/8991", strict=True)
+# @pytest.mark.xfail(reason="https://github.com/dask/dask/issues/8991", strict=True)
```

Comment:
@ian-r-rose - Do you think we should remove the xfail here, or just remove the test altogether?

Reply:
I'm torn on this -- I'd love to have some test that probes whether `key_dependencies` are correct, but I really don't like this one. Do you have any ideas for how to improve it?

Reply:
What I don't like about this is that it is pretty far from user code, and calls a bunch of private methods.

> Maybe we really don't need this to be in test_distributed.py? That is, in test_layers.py we can just test that calling cull on various layers does not change the output of graph.get_all_dependencies()?

I like this idea better. In my memory, the last time I tried this I ran into problems with the fact that actual dependency logic happens in the pack/unpack process. But I could have that wrong, I haven't looked in detail recently.

Reply:
>I ran into problems with the fact that actual dependency logic happens in the pack/unpack process. But I could have that wrong, I haven't looked in detail recently.

Yes - The **actual** dependency logic does happen in the unpack process. However, when `key_dependencies` is empty, `get_all_dependencies` will manually crawl through the materialized graph and use `keys_in_tasks` to construct the "ground truth" for `key_dependencies`. Therefore, it think it should be a good way to test that `HighLevelGraph.cull` isn't tainting the result.